pkg/proxy/proxy.go

```diff
@@ -240,6 +243,21 @@ func unlock(name string) {
 	appMutex[name].Unlock()
 }
 
+// Returns a timeout based on the value of a environment variable.
+// defaults to 300s (5min)
+func getTimeout(envVar string) int64 {
+	timeout := defaultTimeoutSeconds
+	var err error
+	if envTimeout := os.Getenv(envVar); envTimeout != "" {
+		timeout, err = strconv.Atoi(envTimeout)
+		if err != nil {
+			log.Errorf("Unable to use custom timeout. Failed to parse %s as int: %v", envVar, err)
```

Comment:
Avoid failures on purpose, if the `timeout` is not properly set, fallback to the default.